File: tests/testthat/dummy_score.py
# Copyright (c) Microsoft Corporation. All rights reserved.
# Licensed under the MIT License.

from azureml.core.model import Model


def init():
    global model


def run(data):
    try:
        result = model.predict(data)
        # you can return any datatype as long as it is JSON-serializable
        return result.tolist()
    except Exception as e:
        error = str(e)
        return error

File: tests/testthat/dummy_score.R
# Copyright (c) Microsoft Corporation. All rights reserved.
# Licensed under the MIT License.

init <- function()
{
  model <<- readRDS("score.R")
  message("model is loaded")
  
  function(data)
  {
    plant <- as.data.frame(fromJSON(data))
    prediction <- predict(model, plant)
    result <- as.character(prediction)
    toJSON(result)
  }
}